### ISSUE TITLE
Support for SpringData MongoRepository and MongoTemplate

### ISSUE THREAD

**Issue by Frintrop** [quarkusio/quarkus#6508]
Please add support for SpringData´s MongoRepository and MongoRepository.
Companies can reuse existing codebase.

**Comment by Frintrop** [comment]
MongoTemplate seemed to be working for quarkus,
but we have problems with repositories ....
"Currently only Entities with the @Id annotation are supported." is the error ...

**Comment by geoand** [comment]
Yes, we haven't done any work to get such a feature working. But it's certainly something we are considering

**Comment by Frintrop** [comment]
can we support. we are new to mongo-spring and quarkus ...

**Comment by Frintrop** [comment]
if there is a documentation how to make mongo-spring-data a plugin for quarkus

**Comment by geoand** [comment]
Hi,

If you would like to engadge in this work, then I suggest first reading https://quarkus.io/guides/writing-extensions. Then looking (and following along with a debugger) the code of the spring-data-jpa extension would make most sense

**Comment by Frintrop** [comment]
mhh i saw this extension. but u think my experience is too low for that..

**Comment by geoand** [comment]
Perhaps first takling some easier Quarkus issues to gain some experience would make sense?

**Comment by aureamunoz** [comment]
cc @aureamunoz
